tla-pp/src/test/scala/at/forsyte/apalache/tla/pp/TestInliner.scala

```diff
@@ -0,0 +1,349 @@
+package at.forsyte.apalache.tla.pp
+
+import at.forsyte.apalache.tla.lir.TypedPredefs._
+import at.forsyte.apalache.tla.lir._
+import at.forsyte.apalache.tla.lir.convenience.tla
+import at.forsyte.apalache.tla.lir.oper.ApalacheOper
+import at.forsyte.apalache.tla.lir.transformations.impl.IdleTracker
+import at.forsyte.apalache.tla.pp.Inliner.emptyScope
+import org.junit.runner.RunWith
+import org.scalatest.BeforeAndAfterEach
+import org.scalatest.funsuite.AnyFunSuite
+import org.scalatestplus.junit.JUnitRunner
+
+@RunWith(classOf[JUnitRunner])
+class TestInliner extends AnyFunSuite with BeforeAndAfterEach {
```

Comment:
I think it's helpful to have all these unit tests. But we should also introduce comprehensive integration tests for the inliner, similar to `TestSets.tla` and `TestSequences.tla`.

Reply:
Since inlining is a fundamental transformation the entire collection of integration tests also acts as tests for inlining. What would a `TestInlining.tla` even look like?

Reply:
It may duplicate the unit tests and richer combinations of inlining. Tests in TLA+ are way easier to read than the tests in IR.